pageserver/src/tenant.rs

```diff
@@ -1012,41 +1012,30 @@ impl Tenant {
             std::fs::read_dir(&timelines_dir).context("list timelines directory for tenant")?
         {
             let entry = entry.context("read timeline dir entry")?;
-            let timeline_dir = entry.path();
+            let timeline_dir = Utf8PathBuf::from_path_buf(entry.path()).expect("non-Unicode path");
 
             if crate::is_temporary(&timeline_dir) {
-                info!(
-                    "Found temporary timeline directory, removing: {}",
-                    timeline_dir.display()
-                );
+                info!("Found temporary timeline directory, removing: {timeline_dir}",);
                 if let Err(e) = std::fs::remove_dir_all(&timeline_dir) {
-                    error!(
-                        "Failed to remove temporary directory '{}': {:?}",
-                        timeline_dir.display(),
-                        e
-                    );
+                    error!("Failed to remove temporary directory '{timeline_dir}': {e:?}",);
                 }
             } else if is_uninit_mark(&timeline_dir) {
                 if !timeline_dir.exists() {
-                    warn!(
-                        "Timeline dir entry become invalid: {}",
-                        timeline_dir.display()
-                    );
+                    warn!("Timeline dir entry become invalid: {timeline_dir}",);
                     continue;
                 }
 
                 let timeline_uninit_mark_file = &timeline_dir;
                 info!(
-                    "Found an uninit mark file {}, removing the timeline and its uninit mark",
-                    timeline_uninit_mark_file.display()
+                    "Found an uninit mark file {timeline_uninit_mark_file}, removing the timeline and its uninit mark",
                 );
-                let timeline_id = TimelineId::try_from(timeline_uninit_mark_file.file_stem())
-                    .with_context(|| {
-                        format!(
-                            "Could not parse timeline id out of the timeline uninit mark name {}",
-                            timeline_uninit_mark_file.display()
+                let timeline_id =
+                    TimelineId::try_from(timeline_uninit_mark_file.as_std_path().file_stem())
```

Comment:
There is also https://docs.rs/camino/latest/camino/struct.Utf8Path.html#method.file_stem so I don't see why this would require the conversion?

Reply:
Um, currently, only `TryFrom<Option<&OsStr>>` is implemented for `TimelineId`.
That's why I added the conversion.
But, maybe, we can add/update the `TryFrom` impl for `TimelineId`, and use the `Utf8Path::file_stem()` in above case.

Reply:
I added the impl of `TryFrom<Option<&str>>` for `TimelineId`.
So, we can use the `Utf8Path::file_stem()` in above case.